frame/contracts/src/lib.rs

```diff
@@ -924,9 +920,38 @@ pub mod pallet {
 		StorageValue<_, DeletionQueueManager<T>, ValueQuery>;
 }
 
+/// The type of origins supported by the contracts pallet.
+#[derive(Clone, Encode, Decode, PartialEq, TypeInfo, RuntimeDebugNoBound)]
+pub enum Origin<T: Config> {
+	Root,
+	Signed(T::AccountId),
+}
+
+impl<T: Config> Origin<T> {
+	/// Creates a new Signed Caller from an AccountId.
+	pub fn from_account_id(account_id: T::AccountId) -> Self {
+		Origin::Signed(account_id)
+	}
+	/// Creates a new Origin from a OriginFor
+	pub fn from_origin_for(o: OriginFor<T>) -> Result<Self, DispatchError> {
```

Comment:
I know this (and the above `from_account_id`) look ugly and would ideally be a `impl TryFrom/From for Origin`. Something like this probalby:
```rust
impl<T: Config> TryFrom<OriginFor<T>> for Origin<T> {
	type Error = DispatchError;
	fn try_from(o: OriginFor<T>) -> Result<Self, Self::Error> {
		match o.into() {
			Ok(RawOrigin::Root) => Ok(Self::Root),
			Ok(RawOrigin::Signed(t)) => Ok(Self::Signed(t)),
			_ => Err(BadOrigin.into()),
		}
	}
}
```

But the compiler would complain about it:
```
conflicting implementations of trait `TryFrom<_>` for type `Origin<_>`
```

I have discussed it with Alex already and we didn't find a way around the error. If the reviewer has a better idea, I'm open to suggestions.